workflow-templates/im-test-testng-schedule.yml

```
@@ -0,0 +1,173 @@
+
+env:
+  GITHUB_REF: 'main' # Default branch to run against.
+  JAVA_VERSION: '21' # TODO: Update this to the version of Java required for the tests.
+  # Steps to modify the maven settings are available at   https://github.com/im-practices/techhub/blob/main/docs/Azure/content/Health-Checks/TestNG-UI-Health-Check.md Step no 4,5
+  MAVEN_SETTINGS_PATH: './path/to/setup-custom-maven.sh' # TODO: Update this to the Maven setup script.
+  MAVEN_BIN: 'embedded/maven/bin/mvn' # TODO: Update only if you are modifying the path at shell script.
+  MAVEN_CACHE_PATH: '~/.m2/repository' # Maven Repo path.
+  JAVA_AGENT: 'resources/app-insights/applicationinsights-agent.jar' # # TODO: Update this to the App Insights Java agent JAR location.
+  ENV_TEST_RUN_NAME: '' # TODO: Update this with the name of the health check test run.
+  JAVA_TEST_DIR: '' # TODO: Update this to the name of Java test directory.
+  TESTNG_RESULTS_FILE: '' # TODO: Update this to the TestNG results file.
+  TESTNG_SCREENSHOTS_DIR: '' # TODO: Update this to the path of screenshots directory.
+  TESTNG_REPORT_DIR: '' # TODO: Update this to the path to TestNG reports directory.
+
+jobs:
+  maven-cache:
+    # Sets up the Maven cache to speed up dependency installation
+    runs-on: im-linux
+
+    defaults:
+      run:
+        shell: bash
+
+    outputs:
+      MAVEN_CACHE_KEY: ${{ env.MAVEN_CACHE_KEY }}
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ env.GITHUB_REF }}
+
+      - name: Setup Java
+        uses: actions/setup-java@v4
+        if: steps.has-cache.outputs.cache-hit != 'true'
+        with:
+          distribution: 'temurin'
+          java-version: ${{ env.JAVA_VERSION }}
+          
+      - name: Set up Maven Cache Keys
+         # Generate a cache key based on the pom.xml file. If it changes, a new cache will be created.
+        run: echo "MAVEN_CACHE_KEY=maven_repo-${{ hashFiles('**/pom.xml') }}" >> $GITHUB_ENV
+
+      - name: Check for Maven Cache
+        id: has-cache
+        uses: actions/cache@v4
+        with:
+           lookup-only: true
+           key: ${{ env.MAVEN_CACHE_KEY }}
+           path: ${{ env.MAVEN_CACHE_PATH }} # Cache path for Maven dependencies.
+
+      - name: Set up Maven
+        if: steps.has-cache.outputs.cache-hit != 'true'
+        # Steps to generate the shell script is available at  https://github.com/im-practices/techhub/blob/main/docs/Azure/content/Health-Checks/TestNG-UI-Health-Check.md Step no 6
+        # This step will run the shell script to setup Maven with the modified settings.xml file.
+        run: |
+          echo "Setting up Maven and installing dependencies"
+          chmod +x ${{ env.MAVEN_SETTINGS_PATH }}
+          ${{ env.MAVEN_SETTINGS_PATH }}
+        shell: bash
+
+      - name: Build Project using Maven 
+        if: steps.has-cache.outputs.cache-hit != 'true'
+        run: ${{ env.MAVEN_BIN }} clean install
+
+
+  testng-tests:
```

```suggestion
  selenium-maven-tests:
```
It is good practice to name the job after the type of test